### PR TITLE
Feature/debugging update

### PR DESCRIPTION
**Pull request by SaladDays831**
Hey hey hey 
Another cookbook chapter update, this time about `debugging`

Grab some tea, make yourself comfortable, give it a read, write some feedback, you know the drill 👐

Thanks @romanfurman6 for the support and for making this possible 🙏

### CODE REVIEW

File: 10-debugging.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+- Search the project efficiently
+
+You can use Regex in the search bars, but also you can insert patterns with ⌃⌥⌘P like, for example, searching for force unwraps in the whole project
+
+<img src="resources/debugging/search1.png" style="zoom:50%;" />
+
+Also, you can use scopes to your advantage, for example, by excluding third-party libraries, or using `File name` -> `Contains` -> "ViewController" to search only your ViewControllers (if you follow naming conventions, of course)
+
+- Harness the simulator
```
Comment (MykhailoPalchuk) [inline 10-debugging.md:34]: `Harness` is a pretty rare word, I'd suggest using word `exploit`, `enslave` (if you're into it), or to be as simple as possible just `Use full potential of the simulator`

**Review by MykhailoPalchuk — COMMENTED** [review]
Great read! Thank you so much for putting this together! I have 2 comments and a suggestion.
I suggest mentioning Reveal app in debugging tools. It's pretty cool, it's main benefit is that it doesn't freeze your app and Xcode. It also better points at broken constraints and lets you change them on-the-go and observe results. It definitely deserves a mention!

File: 10-debugging.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+### 1. Best Practices
+
+Most of the time we work with code - we're debugging something. It's useful to do that efficiently to save our time and  mental health.
+
+- Search the project efficiently
+
+You can use Regex in the search bars, but also you can insert patterns with ⌃⌥⌘P like, for example, searching for force unwraps in the whole project
+
+<img src="resources/debugging/search1.png" style="zoom:50%;" />
```
[the remark this answers is not in the document]
Comment (SaladDays831) [reply 10-debugging.md:30]: It was actually looking for all exclamation marks and I didn't notice that before. Mentioned different, more useful examples. Thanks!

File: 10-debugging.md (hunk 1 above)
Comment (SaladDays831) [reply 10-debugging.md]: Agree, changed!

**Comment by SaladDays831** [comment]
Thanks @MykhailoPalchuk!
Added a couple of words about Reveal
Had some second thoughts about it earlier because of the paid version, but it seems too cool to leave it out.

**Review by Pan-TERRA — APPROVED** [review]
Awesome work! Thank you!

**Review by romanfurman6 — APPROVED** [review]
Good job man🚀

**Review by vovabndr — APPROVED** [review]

**Comment by SaladDays831** [comment]
Added a couple of words about MetricKit

Thanks @ANDaleksei!

**Review by ANDaleksei — APPROVED** [review]

**Review by MykhailoPalchuk — APPROVED** [review]

**Review by AlexanderKulyk — APPROVED** [review]
Great job 👏 
Huge thanks for improving our cookbook!

**Pull request merged**